### PR TITLE
Make punctuation-only texts inapplicable in R69

### LINKED ISSUE #1166: SIA-R69: Consider making non-text content not applicable
Decorative text like `|` , `&`, etc. should not be flagged as an issue.

### PR DESCRIPTION
**Pull request by rcj-siteimprove**
Resolves #1166

We just ignore text that only consists of punctuation.

### CODE REVIEW

File: packages/alfa-rules/src/sia-r69/rule.ts [hunk 1]
```diff
@@ -23,7 +26,9 @@ export default Rule.Atomic.of<
   evaluate({ device, document }) {
     return {
       applicability() {
-        return nonDisabledTexts(document, device);
+        return nonDisabledTexts(document, device).filter(
+          not(isOnlyPunctuation),
+        );
```
Comment (Jym77) [inline packages/alfa-rules/src/sia-r69/rule.ts:31]:
```suggestion
        return nonDisabledTexts(document, device).reject(
          isOnlyPunctuation,
        );
```
should work.

File: packages/alfa-rules/test/sia-r69/rule.spec.tsx [hunk 2]
```diff
@@ -395,6 +395,40 @@ test(`evaluate() is inapplicable to the text that is part of a label of a disabl
   t.deepEqual(await evaluate(R69, { document }), [inapplicable(R69)]);
 });
 
+test("evaluate() is inapplicable to text that would otherwise pass if it was not only punctuation", async (t) => {
+  const target = h.text(",./;'[]\\-=?<>:\"{}|_+!@#$%^&*()");
+
+  const document = h.document([
+    <html
+      style={{
+        backgroundImage: "url('foo.png')",
```
Comment (Jym77) [inline packages/alfa-rules/test/sia-r69/rule.spec.tsx:404]: **Suggestion:** That technically would make it a CantTell if it was applicable. `backgroundColor` would match the description better.

File: packages/alfa-rules/test/sia-r69/rule.spec.tsx [hunk 3]
```diff
@@ -395,6 +395,40 @@ test(`evaluate() is inapplicable to the text that is part of a label of a disabl
...
+  t.deepEqual(await evaluate(R69, { document }), [inapplicable(R69)]);
+});
+
+test("evaluate() is inapplicable to text that would otherwise fail if it was not only punctuation", async (t) => {
+  const target = h.text(",./;'[]\\-=?<>:\"{}|_+!@#$%^&*()");
+
+  const document = h.document([
+    <html
+      style={{
+        backgroundImage: "url('foo.png')",
```
Comment (Jym77) [inline packages/alfa-rules/test/sia-r69/rule.spec.tsx:421]: Same here. Which gets even stranger since the test case is now exactly the same.

**Review by Jym77 — CHANGES_REQUESTED** [review]

**Comment by Jym77** [comment]
@dan-tripp-siteimprove FYI.

File: packages/alfa-rules/test/sia-r69/rule.spec.tsx (hunk 3 above)
Comment (rcj-siteimprove) [reply packages/alfa-rules/test/sia-r69/rule.spec.tsx]: I based the tests on two other tests that were passing and failing resp., but I forgot to copy the oracle as well. Now I have based the tests on two other tests that are not using an oracle at all to keep it simple. I've confirmed that adding a single letter to the text in both tests results in passing and failing outcome resp.

File: packages/alfa-rules/test/sia-r69/rule.spec.tsx [hunk 4]
```diff
@@ -418,7 +418,8 @@ test("evaluate() is inapplicable to text that would otherwise fail if it was not
   const document = h.document([
     <html
       style={{
-        backgroundImage: "url('foo.png')",
+        backgroundImage: "linear-gradient(#000 50%, transparent 50%)",
+        backgroundColor: "#000",
         color: "#000",
```
Comment (Jym77) [inline packages/alfa-rules/test/sia-r69/rule.spec.tsx:423]: Wouldn't that be inapplicable as invisible because it's black-on-black?

**Review by Jym77 — APPROVED** [review]

File: packages/alfa-rules/test/sia-r69/rule.spec.tsx (hunk 4 above)
Comment (rcj-siteimprove) [reply packages/alfa-rules/test/sia-r69/rule.spec.tsx:423]: For the record, this is answered in above answer

**Comment by dan-tripp-siteimprove** [comment]
Hi @rcj-siteimprove - today I noticed that this PR did it for R69 and not R66.  [Test page here](https://dan-tripp-siteimprove.github.io/contrast-pipe-char-test-page--2025-01-10--QHBRssM.html).  IMO R66 should get the same treatment that this PR gave to R69, because WCAG uses the same language (text = "expressing something in human language") for both success criteria.  If you agree, may I submit a PR to this effect?  :)

**Comment by Jym77** [comment]
Yes R66 should have the same treatment. Sorry we missed that 🙈 
We should move that to the common Applicability, I guess.

**Comment by dan-tripp-siteimprove** [comment]
Cool.  I would like to submit a PR for this.   (Though I'm not sure when.)  Thank you both.

**Pull request merged**